hawtio-util/src/test/java/io/hawt/util/FilesTest.java

```diff
@@ -0,0 +1,301 @@
+    when(target.getParentFile()).thenReturn(targetParent);
+
+    // Act
+    thrown.expect(IOException.class);
+    Files.copy(source, target);
+    // Method is not expected to return due to exception thrown
+  }
+
+  @PrepareForTest({IOHelper.class, Files.class})
+  @Test
+  public void copy() throws Exception, IOException {
+    // Setup mocks
+    final File source = mock(File.class);
+    final File target = mock(File.class);
+    final File targetParent = mock(File.class);
+    final FileInputStream is = mock(FileInputStream.class);
+    final FileOutputStream os = mock(FileOutputStream.class);
+    mockStatic(IOHelper.class);
+
+    // Arrange & Act
+    when(source.exists()).thenReturn(true);
+    when(target.getParentFile()).thenReturn(targetParent);
+    whenNew(FileInputStream.class).withParameterTypes(File.class).withArguments(isA(File.class)).thenReturn(is);
+    whenNew(FileOutputStream.class).withParameterTypes(File.class).withArguments(isA(File.class)).thenReturn(os);
+
+    when(target.exists()).thenReturn(true);
+    when(targetParent.exists()).thenReturn(false);
+    when(targetParent.mkdirs()).thenReturn(false);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+
+    when(target.exists()).thenReturn(false);
+    when(targetParent.mkdirs()).thenReturn(true);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+    
+    when(targetParent.exists()).thenReturn(true);
+    Files.copy(source, target);
+    // Method returns void, testing that no exception is thrown
+  }
+
+  @Test
+  public void recursiveDelete() throws Exception {
+    // Setup mocks
+    final File file = mock(File.class);
+    final File file1 = mock(File.class);
+    final File file21 = mock(File.class);
+    final File file22 = mock(File.class);
+    final File file2 = mock(File.class);
+    final File file3 = mock(File.class);
+
+    // Arrange, Act & Assert result
+    when(file.isDirectory()).thenReturn(false);
+    when(file.delete()).thenReturn(false);
+    Assert.assertEquals(0, Files.recursiveDelete(file));
+
+    when(file.delete()).thenReturn(true);
+    Assert.assertEquals(1, Files.recursiveDelete(file));
+
+    when(file.isDirectory()).thenReturn(true);
+    when(file.listFiles()).thenReturn(null);
+    Assert.assertEquals(1, Files.recursiveDelete(file));
+
+    when(file1.isDirectory()).thenReturn(false);
+    when(file1.delete()).thenReturn(true);
+    when(file21.isDirectory()).thenReturn(false);
+    when(file21.delete()).thenReturn(true);
+    when(file22.isDirectory()).thenReturn(false);
+    when(file22.delete()).thenReturn(true);
+    when(file2.listFiles()).thenReturn(new File[] {file21, file22});
+    when(file2.isDirectory()).thenReturn(true);
+    when(file2.delete()).thenReturn(true);
+    when(file3.isDirectory()).thenReturn(false);
+    when(file3.delete()).thenReturn(true);
+    when(file.isDirectory()).thenReturn(true);
```

Comment:
I think it might be clearer to put all of the `.delete()` mocking setup lines into a separate block, so the test reads like
1. create the mock `File` objects
2. setup the mock directory tree
3. mock all of the `delete` methods
P.S. do you know if it's possible to mock all `delete` calls (in all mocks) with one statement? I'm imagining something like `when(File.delete())...` - like mockStatic but for non-static methods.